pkg/vfs/download_store.go

```diff
@@ -0,0 +1,128 @@
+package vfs
+
+import (
+	"encoding/hex"
+	"time"
+
+	"github.com/cozy/cozy-stack/pkg/crypto"
+)
+
+// A DownloadStore is essentially an object to store Archives & Files by keys
+type DownloadStore interface {
+	AddFile(f string) (string, error)
+	AddArchive(a *Archive) (string, error)
+	GetFile(k string) (string, error)
+	GetArchive(k string) (*Archive, error)
+}
+
+type fileRef struct {
+	Path      string
+	ExpiresAt time.Time
+}
+
+// downloadStoreTTL is the time an Archive stay alive
+const downloadStoreTTL = 1 * time.Hour
+const downloadStoreCleanInterval = 1 * time.Hour
+
+var storeStore map[string]*memStore
```

Comment:
`maps` are not thread safe, you should have a mutex around this one since it is used concurrently